File: 프로그래머스/lv2/42584. 주식가격/주식가격.js
function solution(prices) {
    
    const n = prices.length;
    const result = new Array(n).fill(0);
    const stack = [];
    
    for (let i = 0; i < n; i++) {
        while (stack.length > 0 && prices[i] < prices[stack[stack.length - 1]]) {
            const j = stack.pop();
            result[j] = i - j;
        }
        stack.push(i);
    }

    while (stack.length > 0) {
        const j = stack.pop();
        result[j] = n - 1 - j;
    }
    
    return result;
}

File: 프로그래머스/2/42584. 주식가격/주식가격.py
def solution(prices):
    
    stack = []
    answer = [0] * len(prices)
    
    for i in range(len(prices)):
        while stack and prices[stack[-1]] > prices[i]:
            p = stack.pop()
            answer[p] = i - p
        stack.append(i)
        
    for i in stack:
        answer[i] = len(prices) - 1 - i
    
    return answer